src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -669,213 +355,347 @@ private void ProcessWSManProtocolForTestConnection()
-                        }
-                    }
-                    else
-                    {
-                        ArgumentException Ex = new ArgumentException(StringUtil.Format(ComputerResources.NotEnabled, sysdrive));
-                        WriteError(new ErrorRecord(Ex, "EnableComputerRestoreNotEnabled", ErrorCategory.InvalidOperation, null));
-                    }
-                }
-                catch (ManagementException e)
-                {
-                    if ((e.ErrorCode.Equals(ManagementStatus.NotFound)) || (e.ErrorCode.Equals(ManagementStatus.InvalidClass)))
-                    {
-                        ErrorRecord er = new ErrorRecord(new ArgumentException(StringUtil.Format(ComputerResources.NotSupported)), null, ErrorCategory.InvalidOperation, null);
-                        WriteError(er);
-                    }
-                    else
-                    {
-                        ErrorRecord errorRecord = new ErrorRecord(e, "GetWMIManagementException", ErrorCategory.InvalidOperation, null);
-                        WriteError(errorRecord);
-                    }
-                }
-                catch (COMException e)
-                {
-                    if (string.IsNullOrEmpty(e.Message))
-                    {
-                        Exception Ex = new ArgumentException(StringUtil.Format(ComputerResources.SystemRestoreServiceDisabled));
-                        WriteError(new ErrorRecord(Ex, "ServiceDisabled", ErrorCategory.InvalidOperation, null));
-                    }
-                    else
-                    {
-                        ErrorRecord errorRecord = new ErrorRecord(e, "COMException", ErrorCategory.InvalidOperation, null);
-                        WriteError(errorRecord);
-                    }
-                }
-            }
-            else
+        /// <summary>
+        /// Wait for the PowerShell to be ready
+        /// </summary>
+        PowerShell = 0x2,
+    }
+
+    /// <summary>
+    /// Restarts the computer
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Restart, "Computer", SupportsShouldProcess = true, DefaultParameterSetName = DefaultParameterSet,
+        HelpUri = "https://go.microsoft.com/fwlink/?LinkID=135253", RemotingCapability = RemotingCapability.OwnedByCommand)]
+    public class RestartComputerCommand : PSCmdlet, IDisposable
+    {
+#region "Parameters and PrivateData"
+
+        private const string DefaultParameterSet = "DefaultSet";
+
+        /// <summary>
+        /// The authentication options for CIM_WSMan connection
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateSet(
+            "Default",
+            "Basic",
+            "Negotiate", // can be used with and without credential (without -> PSRP mapped to NegotiateWithImplicitCredential)
+            "CredSSP",
+            "Digest",
+            "Kerberos")] // can be used with and without credential (not sure about implications)
```

Comment:
Comment should be udpated: can be used with explicit or implicit credential

Reply:
fixed